sklearn/utils/multiclass.py

```diff
@@ -316,7 +316,11 @@ def type_of_target(y, input_name=""):
 
     # Invalid inputs
     if y.ndim > 2 or (y.dtype == object and len(y) and not isinstance(y.flat[0], str)):
```

Comment:
```suggestion
    if y.ndim > 2:
        return ("The dimensions of the target cannot be greater than 2.")
    if (y.dtype == object and len(y) and not isinstance(y.flat[0], str)):
        return ("Unknown label type. Use an object array containing strings or encode the values in a contiguous numerical array.")
```

Reply:
We can display a more informative error message by doing this and then try to update the tests accordingly? WDYT @lucyleeow